Port MASTG-TEST-0056 to v2: Determining Whether Sensitive Data Is Exposed via IPC Mechanisms (ios) (#3793)
OWASP/mastg: knowledge/ios/MASVS-PLATFORM/MASTG-KNOW-0078.md

title: Inter-Process Communication (IPC)
---

> iOS does not provide a general-purpose mechanism for arbitrary third-party apps to communicate directly. Instead, apps exchange data through platform-mediated channels, user actions, shared entitlements, or network interfaces.

Reviewer: ```suggestion
iOS does not provide a general-purpose mechanism for third-party apps to communicate directly. Instead, apps exchange data through platform-mediated interfaces of varying levels of abstraction.
```

> Each [IPC mechanism](https://developer.apple.com/forums/tags/inter-process-communication) can be characterized by who can send data, who can receive data, whether user interaction is required, how long the data remains available, and whether the channel is restricted by an entitlement or app group.

Reviewer: nit: You might wanna remove this link IMHO, it links to Apple's forums with the IPC tag which is not very active.
